backend.go

```diff
@@ -184,6 +190,10 @@ func (b *azureSecretBackend) getClient(ctx context.Context, s logical.Storage) (
 		return nil, err
 	}
 
+	if config == nil {
+		return nil, fmt.Errorf("config is nil")
+	}
+
 	if b.settings == nil {
 		if config == nil {
 			config = new(azureConfig)
```

Comment:
Looks like this block will never be triggered if we're guarding against `config == nil` on line 193. 

Perhaps we could move the check on line 193 down below the `if b.settings == nil` block?